File: CanvasScene/CanvasScene/Entities/FilterParams.cs
using System;

namespace CanvasScene
{
    public class FilterParams
    {
        public readonly static FilterParams Empty = new FilterParams();

        public string Name { get; set; }

        /// <summary>
        /// включительно
        /// </summary>
        public int? WidthMin { get; set; }

        /// <summary>
        /// включительно
        /// </summary>
        public int? WidthMax { get; set; }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(this, obj))
            {
                return true;
            }

            if (!(obj is FilterParams another))
            {
                return false;
            }

            return WidthMin == another.WidthMin
                && WidthMax == another.WidthMax
                && string.Equals(Name, another.Name, StringComparison.OrdinalIgnoreCase);
        }

        public override int GetHashCode()
        {
            return WidthMin.GetValueOrDefault(0) ^ WidthMax.GetValueOrDefault(0);
        }

    }
}

File: CanvasScene/CanvasScene/ClientApp/src/app/model/filterParams.ts
export class FilterParams {
  // public static readonly notSet: number = -1;

  public name: string;
  public widthMin: string;
  public widthMax: string;

  constructor() {
    this.name = '';
    this.widthMin = '';
    this.widthMax = '';
  }

  public getRequest(): string {
    var url: string = '';
    var anyParamsAdded: boolean = false;

    if (this.name != '' && this.name != null) {
      url += `${anyParamsAdded ? '&' : '?'}name=${this.name}`;
      anyParamsAdded = true;
    }

    if (this.widthMin != '' && this.widthMin != null) {
      url += `${anyParamsAdded ? '&' : '?'}widthMin=${this.widthMin}`;
      anyParamsAdded = true;
    }

    if (this.widthMax != '' && this.widthMax != null) {
      url += `${anyParamsAdded ? '&' : '?'}widthMax=${this.widthMax}`;
      anyParamsAdded = true;
    }

    return url;
  }
}
